it/v1/endpoints/CreateAndAmendEmploymentExpensesControllerISpec.scala

```diff
@@ -294,9 +178,109 @@ class CreateAndAmendEmploymentExpensesControllerISpec extends IntegrationBaseSpe
           (SERVICE_UNAVAILABLE, "SERVICE_UNAVAILABLE", INTERNAL_SERVER_ERROR, StandardDownstreamError)
         )
 
-        input.foreach(args => (serviceErrorTest _).tupled(args))
+        val extraTysErrors = Seq(
+          (BAD_REQUEST, "INVALID_CORRELATION_ID", INTERNAL_SERVER_ERROR, StandardDownstreamError),
+          (UNPROCESSABLE_ENTITY, "TAX_YEAR_NOT_SUPPORTED", BAD_REQUEST, RuleTaxYearNotSupportedError)
+        )
+
+        (errors ++ extraTysErrors).foreach(args => (serviceErrorTest _).tupled(args))
       }
     }
   }
 
+  private trait Test {
+
+    val nino: String = "AA123456A"
+
+    def taxYear: String
+
+    val amount: BigDecimal = 123.12
+
+    def requestBodyJson: JsValue = Json.parse(
+      s"""
+         |{
+         |    "expenses": {
+         |        "businessTravelCosts": $amount,
+         |        "jobExpenses": $amount,
+         |        "flatRateJobExpenses": $amount,
+         |        "professionalSubscriptions": $amount,
+         |        "hotelAndMealExpenses": $amount,
+         |        "otherAndCapitalAllowances": $amount,
+         |        "vehicleExpenses": $amount,
+         |        "mileageAllowanceRelief": $amount
+         |    }
+         |}
+         |""".stripMargin
+    )
+
+    val hateoasResponse: JsValue = Json.parse(s"""
+                                                 |{
+                                                 |  "links": [
+                                                 |    {
+                                                 |      "href": "/individuals/expenses/employments/$nino/$taxYear",
+                                                 |      "method": "GET",
+                                                 |      "rel": "self"
+                                                 |    },
+                                                 |    {
+                                                 |      "href": "/individuals/expenses/employments/$nino/$taxYear",
+                                                 |      "method": "PUT",
+                                                 |      "rel": "amend-employment-expenses"
+                                                 |    },
+                                                 |    {
+                                                 |      "href": "/individuals/expenses/employments/$nino/$taxYear",
+                                                 |      "method": "DELETE",
+                                                 |      "rel": "delete-employment-expenses"
+                                                 |    }
+                                                 |  ]
+                                                 |}
+                                                 |""".stripMargin)
+
+    def uri: String = s"/employments/$nino/$taxYear"
```

Comment:
Can be a `private val`. Also can remove the : String definition

Reply:
perhaps also rename to mtdUri ?

Reply:
I have renamed it to mtdUri. But it does not like it when I make it a private val (or even val for that matter). Is it because it is using an abstract method for the taxYear, and if it is a val the taxYear cannot be overriden?